flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/network/ClientTest.java

```diff
@@ -881,4 +814,54 @@ protected void initChannel(SocketChannel ch) throws Exception {
     private InetSocketAddress getKvStateServerAddress(Channel serverChannel) {
         return (InetSocketAddress) serverChannel.localAddress();
     }
+
+    private static class ChannelDataCollectingHandler extends ChannelInboundHandlerAdapter {
```

Comment:
Why is this one not shareable? As is couldn't we run into the same issue?

Reply:
Because it is stateful and cannot be shared. Otherwise we might associate a message with a wrong `Channel`.